tools/setupResourceScripts/setup_tdr_resources.py

```diff
@@ -1,213 +1,278 @@
+      policy_member={"email": discoverer})
 
 
 def create_snapshots(clients, dataset_name, snapshots, profile_id):
-    snapshot_ids = []
-    for snapshot_to_upload in snapshots:
-        count = snapshot_to_upload.get('count', 1)
-        for i in range(count):
-            snapshot_name = f"{snapshot_to_upload['name']}{i + 1}"
-            snapshot_request = {
-                'name': snapshot_name,
-                'description': snapshot_to_upload['description'],
-                'contents': [{'datasetName': dataset_name,
-                              'mode': 'byFullView'}],
-                'profileId': profile_id
-            }
-            snapshot = wait_for_job(clients, clients.snapshots_api.create_snapshot(snapshot=snapshot_request))
-            print(f"Created snapshot {snapshot_name} with id: {snapshot['id']}")
-            add_snapshot_policy_members(clients, snapshot['id'], snapshot_to_upload)
-            snapshot_ids.append(snapshot['id'])
-    return snapshot_ids
+  snapshot_ids = []
+  for snapshot_to_upload in snapshots:
+    count = snapshot_to_upload.get('count', 1)
+    for i in range(count):
+      snapshot_name = f"{snapshot_to_upload['name']}{i + 1}"
+      snapshot_request = {
+        'name': snapshot_name,
+        'description': snapshot_to_upload['description'],
+        'contents': [{'datasetName': dataset_name,
+                      'mode': 'byFullView'}],
+        'profileId': profile_id
+      }
+      snapshot = wait_for_job(clients,
+        clients.snapshots_api.create_snapshot(snapshot=snapshot_request))
+      print(f"Created snapshot {snapshot_name} with id: {snapshot['id']}")
+      add_snapshot_policy_members(clients, snapshot['id'], snapshot_to_upload)
+      snapshot_ids.append(snapshot['id'])
+  return snapshot_ids
+
+
+def find_dataset_by_name(name):
+  def find_dataset(dataset):
+    return dataset.name==name
+
+  return find_dataset
+
+
+def delete_dataset_if_exists(name, clients):
+  print(f"Checking if dataset {name} exists")
+  datasets = clients.datasets_api.enumerate_datasets()
+  filtered_datasets = list(filter(find_dataset_by_name(name), datasets.items))
+  if len(filtered_datasets) > 0:
+    print(f"Found dataset {name} with ID {filtered_datasets[0].id}")
+    wait_for_job(clients, clients.datasets_api.delete_dataset(filtered_datasets[0].id))
+    print(f"Deleted dataset {filtered_datasets[0].id}")
+
+
+def add_snapshot_builder_settings(clients, id, directory, snapshot_builder_settings_file):
+  with open(
+    os.path.join("files", directory, snapshot_builder_settings_file)) as dataset_schema_json:
+    # In order to use the client API, we need to publish the updated python client library
+    dataset = requests.post(f"{clients.api_client.configuration.host}/api/repository/v1/datasets/{id}/snapshotBuilder/settings", json=json.load(dataset_schema_json), auth=(
+      'bearer', clients.api_client.configuration.access_token))
```

Comment:
Maybe adding a log that the snapshot builder settings were added would be helpful here